developer/src/kmc-model/src/build-trie.ts

```diff
@@ -374,6 +374,11 @@ namespace Trie {
    */
   function addItemToInternalNode(node: InternalNode, item: Entry, index: number) {
     let char = item.key[index];
+    // If an internal node is the proper site for item, it belongs under the
+    // corresponding (sentinel, internal-use) child node signifying this.
+    if(char === null || char === undefined) {
```

Comment:
This is a good catch but I am wondering how these two situations can arise -- i.e. why we need to test for both `null` and `undefined`?